### PR TITLE
Update binaryen

### PR DESCRIPTION
**Pull request by MaxGraey**
- [x] I've read the contributing guidelines

### CODE REVIEW

File: tests/compiler/std/string-casemapping.optimized.wat [hunk 1]
```diff
@@ -2548,6 +2548,8 @@
              i32.eq
              local.get $3
              i32.const 2
+             i32.sub
+             i32.const 0
```
Comment (MaxGraey) [inline tests/compiler/std/string-casemapping.optimized.wat:2552]: This is regression. It seems need revert some simplified arm in ZeroRemover in binaryen

**Comment by MaxGraey** [comment]
It seems this [binaryen v98.0.0-nightly.20201112 version](https://github.com/AssemblyScript/binaryen.js/commit/7a0cdf7a27dea490504909dd8eb1cb473c9c8d6f) cause to significant regression

**Comment by dcodeIO** [comment]
Any idea so far where the size increase comes from?

**Comment by MaxGraey** [comment]
I can only say it's some problem with hash or comparision alghoritm in recent refactoring of `src/ir/ExpressionAnalyzer.cpp` which make blind `duplicate-function-ellimination` pas in very specific situations.

**Pull request closed without merging**